### PR TITLE
Frontend task 17

### PR DESCRIPTION
**Pull request by pedemonte96**

### CODE REVIEW

File: frontend/customerapp/lib/models/cart.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  void empty() {
+    _order.clear();
+    notifyListeners();
+  }
+
+  int countItems() {
+    int numb = 0;
+    _order.forEach((k, v) => numb += v);
```
Comment (rballeba) [inline frontend/customerapp/lib/models/cart.dart:45]: _order.fold(0, (k, v) => t + v); (Reduce pattern :P)

File: frontend/customerapp/lib/models/location.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    double latRad1 = _latitude * pi / 180.0;
+    double latRad2 = pos2.latitude * pi / 180.0;
+    double difLat = (_latitude - pos2.latitude) * pi / 180.0;
+    double difLon = (_longitude - pos2.longitude) * pi / 180.0;
+    double a = sin(difLat / 2) * sin(difLat / 2) +
+        cos(latRad1) * cos(latRad2) * sin(difLon / 2) * sin(difLon / 2);
+    double c = 2 * atan2(sqrt(a), sqrt(1 - a));
+
+    return R * c;
+  }
```
Comment (rballeba) [inline frontend/customerapp/lib/models/location.dart:31]: Very nice!

**Review by rballeba — APPROVED** [review]

File: frontend/customerapp/lib/models/location.dart (hunk 2 above)
Comment (pedemonte96) [reply frontend/customerapp/lib/models/location.dart:31]: :D

File: frontend/customerapp/lib/models/cart.dart (hunk 1 above)
Comment (pedemonte96) [reply frontend/customerapp/lib/models/cart.dart:43]: I don't get this line of code :S

Comment (mariomoliner) [reply frontend/customerapp/lib/models/cart.dart:43]: super nice code marti! good job!

**Review by mariomoliner — APPROVED** [review]

File: frontend/customerapp/lib/models/cart.dart (hunk 1 above)
Comment (pedemonte96) [reply frontend/customerapp/lib/models/cart.dart:43]: Thanks!!! :)

**Pull request closed without merging**